substrate/client/authority-discovery/src/worker.rs

```diff
@@ -668,6 +752,60 @@ where
 		Ok(())
 	}
 
+	fn handle_new_record(
+		&mut self,
+		authority_id: &AuthorityId,
+		kadmelia_key: KademliaKey,
+		new_record: RecordInfo,
+	) -> bool {
+		let current_record_info = self.last_known_record.entry(kadmelia_key.clone()).or_default();
+		if new_record.creation_time > current_record_info.creation_time {
+			let peers_that_need_updating = current_record_info.peers_with_record.clone();
+			for record in new_record.record.iter() {
+				self.network.put_record_to(
```

Comment:
Is it possible to hit a race condition where a 3rd node has already updated the peers from `current_record_info.peers_with_record` with a more up-to-date data after we last received `current_record_info`?

Reply:
Maybe let me think about it.

Reply:
Thought about it for a bit I don't think this should be a problem, because we alway push for the newer record and records get published periodically, all nodes all eventually converge to the newest record.